dataset-spec/README.md

```diff
@@ -0,0 +1,92 @@
+
+*\* There is no standardized name for the concept we are describing here. Others called it: dataset series (ISO 19115), collection (CNES, NASA), dataset (JAXA), dataset series (ESA), product (JAXA).*
+
+## Core
+
+| Element       | Type                                  | Name                            | Description                                                  |
+| ------------- | ------------------------------------- | ------------------------------- | ------------------------------------------------------------ |
+| id            | string                                | Dataset ID (required)           | Identifier for the dataset that is unique across the provider. MUST follow the pattern ` ^[A-Za-z0-9_\-\/]+$ `. TODO: Allow slash? |
+| title         | string                                | Title                           | A short descriptive one-line title for the dataset.          |
+| description   | string                                | Description (required)          | Detailed multi-line description to fully explain the entity. [CommonMark 0.28](http://commonmark.org/) syntax MAY be used for rich text representation. |
+| keywords      | [string]                              | Keywords                        | List of keywords describing the dataset.                     |
+| version       | string                                | Dataset Version                 | Version of the dataset. [Semantic Versioning (SemVer)](https://semver.org/) SHOULD be followed. |
+| license_name  | string                                | Dataset License Name (required) | Dataset's license name based on [SPDX License Identifier](https://spdx.org/licenses/) or `proprietary` (see `license_url`). TODO: How to handle non SPDX? internal, proprietary, ...? |
+| license_url   | string                                | Dataset License URL             | Dataset's license URL SHOULD be specified if `license_name`  does not contain a SPDX License Identifier. |
+| provider      | [Provider Object]                     | Data Provider                   | The organization that creates the content of the dataset.    |
+| host          | Host Object                           | Storage Provider                | The organization that hosts the dataset.                     |
+| geometry      | [GeoJSON Object](http://geojson.org/) | Spatial extent (required)       | The spatial extent covered by the dataset as [GeoJSON](http://geojson.org/) object. |
+| datetime      | string                                | Temporal extent (required)      | Temporal extent covered by the dataset. Date/time intervals MUST be formatted according to ISO 8601. Open date ranges are not supported by ISO 8601 and MUST be encoded as proposed by [Dublin Core Collection Description: Open Date Range Format](http://www.ukoln.ac.uk/metadata/dcmi/date-dccd-odrf/2005-08-13/). |
```

Comment:
I think it could be good here to not reuse datetime and geometry, as those are item specific, and instead really talk about the extent. 

Ideal to me would be to align with WFS3, so we can add the dataset fields directly to their 'collection' metadata. They have it like:

"extent": {
        "spatial": [ 7.01, 50.63, 7.22, 50.78 ],
        "temporal": [ "2010-02-15T12:34:56Z", "2018-03-18T12:11:00Z" ]
      },

See 'example 4' on https://rawgit.com/opengeospatial/WFS_FES/master/docs/17-069.html

I could be up for a different form, like flatten to spatial_extent and temporal_extent - but I don't think we should reuse the item fields, and I don't believe a dataset should need to conform to the Item schema. If we go with something different I'd want to advocate to WFS3 group to align with us. And I'd see a WFS 3 compliant STAC implementation using these dataset fields in their /collections/{collectionId}/ endpoint.

Reply:
I'm fine with renaming it to spatial extent and temporal extent, we just used those names for consistency within STAC. I'll change that in the next minutes. I don't like the the way temporal extents are specified in WFS though as it doesn't allow to leave the end time out (i.e. open date ranges).